lib/model/classycat_classify.py

```diff
@@ -0,0 +1,129 @@
+import httpx
+from openai import OpenAI
+from lib.logger import logger
+from lib.model.model import Model
+from lib.schemas import Message, ClassyCatBatchClassificationResponse
+from lib.s3 import load_file_from_s3, file_exists_in_s3, upload_file_to_s3
+
+
+class LLMClient:
+    def __init__(self):
+        self.client = None
+
+    def get_client(self):
+        pass
+
+    def classify(self, task_prompt, items_count, max_tokens_per_item=50):
+        pass
+
+
+class OpenRouterClient(LLMClient):
+    def __init__(self, model_name):
+        super().__init__()
+        self.model_name = model_name
+
+    def get_client(self):
+        if self.client is None:
+            self.client = OpenAI(base_url='https://openrouter.ai/api/v1',
+                                 api_key=os.environ.get('OPENROUTER_API_KEY'),
+                                 timeout=httpx.Timeout(60.0, read=60.0, write=60.0, connect=60.0), max_retries=0)
+        return self.client
+
+    def classify(self, task_prompt, items_count, max_tokens_per_item=200):
+        client = self.get_client()
+        completion = client.chat.completions.create(
+            model=self.model_name,
+            messages=[
+                {"role": "user", "content": task_prompt}
+            ],
+            max_tokens=(max_tokens_per_item * items_count) + 15,
+            temperature=0.5
+        )
+
+        return completion.choices[0].message.content
+
+
+class Model(Model):
+    def __init__(self):
+        super().__init__()
+        self.output_bucket = os.getenv("CLASSYCAT_OUTPUT_BUCKET")
+        self.batch_size_limit = int(os.environ.get("CLASSYCAT_BATCH_SIZE_LIMIT"))
+        llm_model_name = os.environ.get('CLASSYCAT_LLM_MODEL_NAME')
+        self.llm_client = OpenRouterClient(llm_model_name)
+
+
+    def format_input_for_classification_prompt(self, items):
+        return '\n'.join([f"<ITEM_{i}>{item}</ITEM_{i}>" for i, item in enumerate(items)])
+
+    def parse_classification_results(self, raw_results):
+        results = raw_results.strip()
+        results = results.split('<OUTPUT>')[1].split('</OUTPUT>')[0].strip().split('\n')
+
+        final_results = []
+        for i, result in enumerate(results):
+            final_result = result.split(f'</CATEGORIES_{i}>')[0].split(f'<CATEGORIES_{i}>')[1]
+            final_results.append(final_result.strip().split(';'))
+
+        return final_results
+
+    def classify(self, prompt_template, items, max_tokens_per_item=200):
+        task_prompt = prompt_template.replace('<INSERT_ITEMS_HERE>', self.format_input_for_classification_prompt(items))
+        raw_results = self.llm_client.classify(task_prompt, len(items), max_tokens_per_item)
+        logger.info(f"Raw completion: {raw_results}")
+        return self.parse_classification_results(raw_results)
+
+    def classify_and_store_results(self, schema_id, items):
+        # load prompt from schema
+        schema = load_file_from_s3(self.output_bucket, f"{schema_id}.json")
```

Comment:
Idea: (maybe not relevant if s3 load is small percent of time compared to LLM roundtrip) Since we don't have a huge number of schemas, and not huge, could load them up the first time we see them and not reload from s3 for every classification event?

Reply:
a good idea but probably premature optimization at this point. we should def consider caching schemas if we ever need performance improvements here.

Reply:
From my quick read, that seems like we'd need maybe some global memoized store of these that persists across process iterations? If so that's a fairly substantive departure from previous assumptions about statelessness?

Reply:
totally fair! maybe the best solution would be to cache the schemas, that way we won't have to worry about preserving state or not. but I still think this will not be a bottleneck anytime soon, since per Skye's comments, this is not a substantial part of the request lifecycle (it takes 40+ seconds to hear back from LLMs) and therefore optimizing it would not bring meaningful improvements to the endpoint at this point.

Reply:
Sounds right. Let's not try to optimize the schema loading right now.